### PR TITLE
fix(lsp): hang when caching failed

### LINKED ISSUE #24600: jsxImportSource freezes lsp when deno.json in nested directory is not listed as member of workspace
Version: Deno 1.45.2

Using vscode, if a workspace is set up and there's a `deno.jsonc` with `jsxImportSource` that isn't a member of the workspace, deno language server is unresponsive.

Example directory:
- foo/
  - bar/
    - deno.jsonc
  - baz/
    - deno.jsonc
  - deno.jsonc

foo/deno.jsonc:
```json
{
  "workspace": ["./bar"]
}
```

foo/baz/deno.jsonc:
```json
{
  "compilerOptions": {
    "jsxImportSource": "jsx"
  }
}
```

From there, you can try to make a new typescript file and notice that there's no intellisense. Commenting out the `jsxImportSource` won't fix the issue until the language server is somehow restarted (which fails from the command palette).

<details>
<summary>
Output from Deno Language Server (anonymised)
</summary>

```
Starting Deno language server...
  version: 1.45.2 (release, x86_64-pc-windows-msvc)
  executable: ...\deno.EXE
Connected to "Visual Studio Code" 1.91.1
Enabling import suggestions for: https://deno.land
Refreshing configuration tree...
  Resolved Deno configuration file: "file:///.../foo/deno.jsonc"
  Resolved Deno configuration file: "file:///.../foo/bar/deno.jsonc"
  Resolved Deno configuration file: "file:///.../foo/baz/deno.jsonc"
Error preparing caching: Config file must be a member of the workspace.
  Config: file:///.../foo/baz/deno.jsonc
  Workspace: file:///.../foo
Server ready.
```
</details>

### PR DESCRIPTION
**Pull request by dsherret**
Closes #24600

### CODE REVIEW

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -292,12 +292,7 @@ impl LanguageServer {
         Ok(maybe_cache_result) => maybe_cache_result,
         Err(err) => {
           lsp_warn!("Error preparing caching: {:#}", err);
-          self
-            .inner
-            .read()
-            .await
-            .client
-            .show_message(MessageType::WARNING, err);
+          inner.client.show_message(MessageType::WARNING, err);
```
Comment (dsherret) [inline cli/lsp/language_server.rs:295]: Bug is here

File: cli/lsp/language_server.rs
Comment (dsherret) [inline cli/lsp/language_server.rs:1]: The `LanguageServer` struct has started doing too much stuff increasing the chances of hangs like this. We should push a lot of this back down to be in `Inner`.

Comment (dsherret) [reply cli/lsp/language_server.rs:1]: Actually, it's not too bad. Just seems like within cache we should move more of this down to make this situation very obvious in an eyeshot.

File: cli/lsp/language_server.rs [hunk 2]
```diff
@@ -310,18 +305,13 @@ impl LanguageServer {
...
-          .inner
-          .read()
-          .await
-          .client
-          .show_message(MessageType::WARNING, err);
-      }
 
       // now get the lock back to update with the new information
       let mut inner = self.inner.write().await;
+      if let Err(err) = handle.await.unwrap() {
```
Comment (dsherret) [inline cli/lsp/language_server.rs:311]: Oh wait, this is a bug I just introduced.

Comment (dsherret) [reply cli/lsp/language_server.rs]: Fixed. I updated the code to reduce the chances of this as well.

File: cli/lsp/language_server.rs
Comment (dsherret) [reply cli/lsp/language_server.rs:1]: Updated the code to push more down.

**Review by nayeemrmn — APPROVED** [review]
LGTM

**Pull request merged**